Instagram/View Controllers/LoginViewController.m

```diff
@@ -12,57 +12,88 @@ @interface LoginViewController ()
 
 @property (weak, nonatomic) IBOutlet UITextField *usernameField;
 @property (weak, nonatomic) IBOutlet UITextField *passwordField;
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
+@property (strong, nonatomic) UIAlertController *blankAlert;
+@property (strong, nonatomic) UIAlertController *registrationAlert;
+@property (strong, nonatomic) UIAlertController *loginAlert;
 
 @end
 
 @implementation LoginViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
+    
+    self.blankAlert = [UIAlertController alertControllerWithTitle:@"Username or password is empty." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+        // handle response here.
+    }];
+    [self.blankAlert addAction:okAction];
+    
+    self.registrationAlert = [UIAlertController alertControllerWithTitle:@"Error registering user." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
+    [self.registrationAlert addAction:okAction];
+    
+    self.loginAlert = [UIAlertController alertControllerWithTitle:@"Error during user login." message:@"Please try again!" preferredStyle:(UIAlertControllerStyleAlert)];
+    [self.loginAlert addAction:okAction];
 }
 - (IBAction)registerUser:(id)sender {
-    // initialize a user object
-    PFUser *newUser = [PFUser user];
-    
-    // set user properties
-    newUser.username = self.usernameField.text;
-    newUser.password = self.passwordField.text;
+    [self.activityIndicator startAnimating];
     
-    // call sign up function on the object
-    [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
-        if (error != nil) {
-            NSLog(@"Error: %@", error.localizedDescription);
-        } else {
-            NSLog(@"User registered successfully");
-            [self performSegueWithIdentifier:@"loginSegue" sender:nil];
-        }
-    }];
+    if([self.usernameField.text isEqual:@""] || [self.passwordField.text isEqual:@""])
+    {
+        [self presentViewController:self.blankAlert animated:YES completion:^{
+            [self.activityIndicator stopAnimating];
```

Comment:
This will actually hide the activityIndicator AFTER the alert pops up. You probably wouldn't notice the timing but something to be aware of.